internal/k8s/kctl/deploy.go

```diff
@@ -33,7 +46,18 @@ func Apply(config Config, plugin plugins.PluginSpec, evt Wait) error {
 
 // Delete removes the plugin from the cluster
 func Delete(config Config, plugin plugins.PluginSpec) error {
-	return handleYaml(config, delete, plugin)
+	for i := len(plugin.Yaml) - 1; i >= 0; i-- {
+		yamlSpec := plugin.Yaml[i]
+		err := execute(config, k3sExec, kubectl, delete,
+			decodeType(yamlSpec.Type), yamlSpec.URL, "-n", plugin.NameSpace)
+		if err != nil {
+			log.Fatalf("Error during delete: %s", err.Error())
+		}
+		pause()
+	}
+	_ = deleteNameSpace(config, plugin.NameSpace)
```

Comment:
On a more closer look, I think we should not delete namespaces. As there could be multiple plugins installed in a namespace.

Reply:
good point. We could check if there are dependencies before remove it.

Reply:
I will remove the `delete` I will open another issue to handle it.